enterprise/server/backends/pebble_cache/pebble_cache.go

```diff
@@ -368,6 +392,43 @@ func NewPebbleCache(env environment.Env, opts *Options) (*PebbleCache, error) {
 		return nil, err
 	}
 	pc.minDBVersion, pc.maxDBVersion = filestore.PebbleKeyVersion(versionMetadata.GetMinVersion()), filestore.PebbleKeyVersion(versionMetadata.GetMaxVersion())
+	if pc.activeDatabaseVersion() < pc.minDBVersion {
+		pc.minDBVersion = pc.activeDatabaseVersion()
+	}
+	if pc.activeDatabaseVersion() > pc.maxDBVersion {
+		pc.maxDBVersion = pc.activeDatabaseVersion()
+	}
+
+	// Update the database version, in case the active version has changed.
+	// This will update the DB min/max version, and ensure old/new data is
+	// correctly seen and written.
+	if err := pc.updateDatabaseVersions(pc.minDBVersion, pc.maxDBVersion); err != nil {
+		return nil, err
+	}
+	log.Infof("Min DB version: %d, Max DB version: %d, Active version: %d", pc.minDBVersion, pc.maxDBVersion, pc.activeDatabaseVersion())
+
+	// Only enable migrators if the data stored in the database lags the
+	// currently active version.
+	if pc.minDBVersion < pc.activeDatabaseVersion() {
+		// N.B. Migrators must be added *in order*.
+		if pc.activeDatabaseVersion() >= filestore.Version1 {
+			// Migrate keys from 0->1.
+			pc.migrators = append(pc.migrators, &v0ToV1Migrator{})
+		}
+		if pc.activeDatabaseVersion() >= filestore.Version2 {
+			// Migrate keys from 1->2.
+			pc.migrators = append(pc.migrators, &v1ToV2Migrator{})
+		}
+	}
+
+	// Check that there is a migrator enabled to update us to (or past) the
+	// activeKeyVersion (flag configured). Warn if not.
+	if len(pc.migrators) > 0 {
+		lastMigratorVersion := pc.migrators[len(pc.migrators)-1].ToVersion()
+		if pc.activeDatabaseVersion() > lastMigratorVersion {
+			log.Fatalf("Cache versions will never converge! Active key version %d > last migrator version: %d", pc.activeDatabaseVersion(), lastMigratorVersion)
```

Comment:
NBD, but this func already has an error return type so you could return an error here and let the caller call log.Fatal